docs/troubleshooting-guide/application-database.md

```diff
@@ -0,0 +1,50 @@
+## Specific Problems:
+
+
+### Metabase fails to start due to database locks
+
+Sometimes Metabase will fail to complete its startup due to a database lock that was not cleared properly. The error message will look something like:
+
+    liquibase.exception.DatabaseException: liquibase.exception.LockException: Could not acquire change log lock.
+
+When this happens, go to a terminal where Metabase is installed and run:
+
+    java -jar metabase.jar migrate release-locks
+
+in the command line to manually clear the locks. Then restart your Metabase instance.
+
+### Metabase H2 application database gets corrupted
+
+Because H2 is an on-disk database, it is sensitive to filesystem errors. Sometimes drives get corrupted, or the file doesn't get flushed correctly, which can result in a corrupted database. In these situations, you'll see errors on startup. These vary, but one example is 
+```
+myUser@myIp:~$ java -cp metabase.jar org.h2.tools.RunScript -script whatever.sql -url jdbc:h2:~/metabase.db
+Exception in thread "main" org.h2.jdbc.JdbcSQLException: Row not found when trying to delete from index """"".I37: ( /* key:7864 */ X'5256470012572027c82fc5d2bfb855264ab45f8fec4cf48b0620ccad281d2fe4', 165)" [90112-194]
+    at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
+    [etc]
+```
+
+Not all H2 errors are recoverable (which is why if you're using H2, _please_ have a backup strategy for the application database file). To attempt to recover a corrupted H2 file, try the below.
+
+```
+java -cp metabase.jar org.h2.tools.Recover
+mv metabase.db.h2.db metabase.old.db
+touch metabase.db.h2.db
+java -cp metabase.jar org.h2.tools.RunScript -script metabase.db.h2.sql -url jdbc:h2:/path/to/metabase.db;MV_STORE=FALSE
+```
+
+
+### Metabase fails to connect to H2 Database on Windows 10
+
+In some situations the Metabase JAR needs to be unblocked so it has permissions to create local files for the application database.
+
+On Windows 10, if you see an error message like
+
+    Exception in thread "main" java.lang.AssertionError: Assert failed: Unable to connect to Metabase DB.
+
+when running the JAR, you can unblock the file by right-clicking, clicking "Properties," and then clicking "Unblock."
+See Microsoft's documentation [here](https://blogs.msdn.microsoft.com/delay/p/unblockingdownloadedfile/) for more details on unblocking downloaded files.
+
+There are a few other reasons why Metabase might not be able to connect to your H2 DB. Metabase connects to the DB over a TCP port, and it's possible
+that something in your `ipconfig` configuration is blocking the H2 port. See the discussion [here](https://github.com/metabase/metabase/issues/1871) for
+details on how to resolve this issue.
```

Comment:
you can also take this last paragraph out since as of 25 or 24 we stopped running H2 in server mode, to fix this specific problem. I just forgot to update the docs.